sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1660,8 +1663,9 @@ object GpuOverrides {
     expr[Max](
       "Max aggregate operator",
       ExprChecks.fullAgg(
-        TypeSig.commonCudfTypes + TypeSig.NULL, TypeSig.orderable,
-        Seq(ParamCheck("input", TypeSig.commonCudfTypes + TypeSig.NULL, TypeSig.orderable))),
+        TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.NULL, TypeSig.orderable,
```

Comment:
If we are going to add DECIMAL for MAX in all cases we need some tests to verify that it will work for groupby and reduction use cases as well.

Reply:
The TypeChecks of `Max` and `Min` has been refined. In the updated version, `DECIMAL` TypeSig is only supported in `WindowAggExprContext`.